kube/pod.go

```diff
@@ -450,8 +459,36 @@ func getSecurityContext(role *model.Role) helm.Node {
 		capabilities = append(capabilities, cap)
 	}
 	if len(capabilities) == 0 {
+		if createHelmChart {
```

Comment:
The `len(capabilities) == 0` chunk is basically only relevant if 1) we're _not_ building helm charts, and 2) there are no capabilities. We don't need to make the helm chart path complicated like this, just `range` over an empty / `nil` list instead.

Reply:
Ok, we have lots of roles with (2). Pretty much all but one.
However I am not sure about (1).  When I am building helm charts I __do__ hit this new code for all these roles with (2). So it looks to be needed.

Reply:
Yes, currently it would hit this code; I'm saying we can change this to:
```golang
if createHelmChart {
	caplist := helm.NewList()
	for _, cap := range capabilities {
		// If there are no capabilities, this just never gets added
		caplist.Add(cap)
	}
	caplist.Add(helm.NewNode("{{ . | upper }}", helm.Block(fmt.Sprintf("range %s", config))))
	cla := helm.NewMapping("add", caplist)
	cla.Set(helm.Block(notAll))
	sc := helm.NewMapping()
	sc.Add("privileged", helm.NewNode(true, helm.Block(hasAll)))
	sc.Add("capabilities", cla)
	return sc
}
if len(capabilities) == 0 {
	return nil
}
return helm.NewMapping("capabilities", helm.NewMapping("add", helm.NewNode(capabilities)))
```
}

Reply:
I believe I see now. Move the block dealing with mode `caplist` to the front. That way it can also deal with mode `nil` also, and the entire specialization for `nil` can be removed. It makes a small difference in that mode `nil` now also uses a range instead of direct assignment of the user's cap-list. Bearable.

Will do.

And sorry, while with re-reading your first comment I am now seeing a glimmer of how you implied this optimization this is definitely hindsight.

Reply:
Sorry, I should have explained it better originally; ended up falling back to code because, that, unlike English, is unambiguous (hopefully). :grin: